typescript/src/lib/bitcoin/csuint.ts

```diff
@@ -0,0 +1,48 @@
+import { Hex } from "../utils"
+
+/**
+ * Reads the leading compact size uint from the provided variable length data.
+ *
+ * WARNING: CURRENTLY, THIS FUNCTION SUPPORTS ONLY 1-BYTE COMPACT SIZE UINTS
+ * AND WILL THROW ON COMPACT SIZE UINTS OF DIFFERENT BYTE LENGTH.
+ *
+ * @param varLenData Variable length data preceded by a compact size uint.
+ * @returns An object holding the value of the compact size uint along with the
+ *          compact size uint byte length.
+ */
+function read(varLenData: Hex): {
```

Comment:
Could it be `HexLike` I mentioned in another comment?

```suggestion
function read(varLenData: HexLike): {
```

Reply:
I'd prefer to not use ether's types as part of our public API directly. Let's stick with our own `Hex`. If we need to replace ethers one day, this decision will be beneficial.